### PR TITLE
[TNH_009]Book_listings

### PR DESCRIPTION
**Pull request by JakeGiangan**
[Commands]
`bundle install`
`rails db:migrate:reset db:seed`
`rails s`

[Pre-conditions]
1.) Create multiple listings using seeded account
2.) Create another user account and book the listing

[Expected Output]
1.) Can now book listings
2.) Listing bookings will check for available dates and disables previous dates and the booked dates

[Notes]
1.) In order to find a Listing you have to search in the Index page's Search Bar
2.) If the Host views their own listing, the booking section of the page is hidden/not visible.. this is to disallow the user to book their own listings
3.) to view your listings bookings and your bookings go to Hosting Dashboard -> Bookings

### CODE REVIEW

File: app/views/bookings/_your_listing_bookings.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<div class="row">
+    <div class="col">
+        <div class="card" style="margin-top: 20px;">
+            <div class="card-header">
+                <h5 class="mb-0">Your Listing Bookings</h5>
+            </div>
+            <div class="card-body">
+                <% @user_listings.each do |item|  %>
+                <div class="row" style="margin-bottom: 15px;">
```
Comment (aplazand20) [inline app/views/bookings/_your_listing_bookings.html.erb]: Fix your indention for each do

File: app/views/bookings/_your_trips.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                    <div class="col-xl-2 d-xl-flex align-items-xl-center">
+                        <p><%= item.check_in_date.strftime("%B %d, %Y") %></p>
+                    </div>
+                    <div class="col-xl-2"><img /></div>
+                    <div class="col d-xl-flex align-items-xl-center">
+                        <p><%= item.listing_name %><br /><%= item.first_name %> <%= item.last_name %></p>
+                    </div>
+                    <div class="col d-xl-flex align-items-xl-center"><button class="btn btn-primary col-lg-12" type="button">Review Host</button></div>
+                </div>
+                <% end %>
```
Comment (aplazand20) [inline app/views/bookings/_your_trips.html.erb]: Fix your indention for each do

File: app/views/bookings/bookings.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                        <%= render 'your_listing_bookings' %>
+                    </div>
+                    <div role="tabpanel" class="tab-pane" id="tab-2">
+                        <%= render 'your_trips' %>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (aplazand20) [inline app/views/bookings/bookings.html.erb]: Fix your indentions in this file

File: app/views/listings/_booking.html.erb [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                <p class="text-center MontserratFont">$ <span id="total_price"></span></p>
+            </div>
+        </div>
+        <div class="row text-center d-xl-flex justify-content-xl-center">
+            <div class="col text-center d-xl-flex justify-content-xl-center">
+                <%= f.submit "RESERVE", id: 'reserve', class: "btn btn-success d-xl-flex roundedButton" %>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (aplazand20) [inline app/views/listings/_booking.html.erb]: Fixy our indentions here

File: app/views/listings/_details.html.erb [hunk 5]
```diff
@@ -79,55 +79,11 @@
...
-                    </div>
-                    <div class="col">
-                        <p class="text-center MontserratFont">$total price</p>
-                    </div>
-                </div>
-                <div class="row text-center d-xl-flex justify-content-xl-center">
-                    <div class="col text-center d-xl-flex justify-content-xl-center"><button class="btn btn-success d-xl-flex roundedButton" type="button" >RESERVE</button></div>
-                </div>
-            </div>
+           <%= render'booking' %>
```
Comment (aplazand20) [inline app/views/listings/_details.html.erb]: Fix your indention for if else

File: app/assets/stylesheets/bookings.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Bookings controller here.
```
Comment (aplazand20) [inline app/assets/stylesheets/bookings.scss]: Remove this file if not needed

**Review by aplazand20 — COMMENTED** [review]
Most of the comments are for indentions. Please double check your code before pushing again 🙇

File: app/helpers/bookings_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module BookingsHelper
```
Comment (charlotteds) [inline app/helpers/bookings_helper.rb]: Delete unused file.

File: app/views/bookings/bookings.html.erb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        </div>
+    </div>
+</div>
```
Comment (charlotteds) [inline app/views/bookings/bookings.html.erb]: Delete extra closing tags

File: app/helpers/bookings_helper.rb (hunk 7 above)
Comment (charlotteds) [inline app/helpers/bookings_helper.rb]: Delete unused file

**Review by charlotteds — COMMENTED** [review]
Unrelated error:
- I got an error when I tried to sign up with another account. Please use `string` for the phone number
image

**Review by aplazand20 — APPROVED** [review]

**Review by charlotteds — APPROVED** [review]

**Pull request merged**